### DISCUSSION TITLE
How can I make validation rules of uploaded files “on fly”  ?

### DISCUSSION THREAD

**Discussion by sergeynilov** [orchidsoftware/platform#2428] [Q&A]
I make validation of uploaded files “on fly” based on data in database.
Before selection of an file/image user selects data of type and depending on this 
selection validation rules can be different.
As Upload component is opened and is visible in preview area before users submits a form
if there is a way to use custom Validation Request for image uploading before selected file is opened in preview area?

Please examples or ref to docs.

```
        "laravel/framework": "^9.19",
        "orchid/crud": "^3.8",
        "orchid/platform": "^13.6",

```
Thanks!

**Comment by tabuna** [comment]
You can use the basic file uploader. 

```php
Input::make('avatar')->type('file')
```

It will pass the file to your method and you can do any checks before saving
